java-checks-test-sources/src/main/java/checks/regex/ReluctantQuantifierCheck.java

```diff
@@ -4,7 +4,21 @@
 
   void noncompliant(String str) {
     str.matches("<.+?>"); // Noncompliant [[sc=19;ec=22]] {{Replace this use of a reluctant quantifier with "[^>]++".}}
+    str.matches("<\\S+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\s]++".}}
+    str.matches("<\\D+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\d]++".}}
+    str.matches("<\\W+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\w]++".}}
+    str.matches("<\\s+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\S]++".}}
+    str.matches("<\\d+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\D]++".}}
+    str.matches("<\\w+?>"); // Noncompliant [[sc=19;ec=24]] {{Replace this use of a reluctant quantifier with "[^>\\W]++".}}
```

Comment:
In these cases (and the later ones involving \s, \d and \w), the suggested replacement is too complicated because those classes don't contain `>`, so there's no need to exclude it. This also means that there's no potential for catastrophic backtracking behavior here, so using the `?` here is not really problematic, but also entirely unnecessary (the regex would behave the same without it and be somewhat more efficient).

So in cases like this there should either be no issue or the message should instead be something along the lines of "Make this regex greedy" or "Remove this '?'".

To check whether a character class overlaps with the character after it, `new SimplifiedRegexCharacterClass(base).intersects(new SimplifiedRegexCharacterClass(escapedClass), true)` can be used in the code.